presto-common/src/main/java/com/facebook/presto/common/block/Block.java

```diff
@@ -400,4 +400,20 @@ default long toLong(int position)
     {
         throw new UnsupportedOperationException(getClass().getName());
     }
+
+    /**
+     * Returns the underlying value block underlying this block.
+     */
+    default Block getUnderlyingValueBlock()
+    {
+        return this;
+    }
+
+    /**
+     * Returns the position in the underlying value block corresponding to the specified position in this block.
+     */
+    default int getUnderlyingValuePosition(int position)
+    {
+        return position;
+    }
```

Comment:
Can you help me understand the rationale for these new methods?  I think they're just used to retrieve a byte, why do we need to unwrap the dictionary to do that?